### PR TITLE
[READY TO MERGE] Bandits Steal the Show: Class Selection for the Bandit Antag.

### PR DESCRIPTION
**Pull request by Plum-but-gayer**
## About The Pull Request
Provides class selection for bandits, giving them six(!!) new classes to choose from, some with subclasses or weapon selection.
Bandits are balanced to now be similar in power level to an adventurer. Looking at the code, they were originally balanced to be in groups of 16 or more. 

Also gives a small update to the bandit camp, adding a few items (spare vulf caps, a healing potion, spare keys, disguise cloaks in a variety of colors so they're not immediately found out) a building (for private RP scenes) and also attempts to remap the forest to make it a little more interesting than the click-drag blocks of duplicate evil tree . jpg. Inspired by some other person's bandit camp revamp PR that I saw once but couldn't find again so I just kinda tried to do it from memory. Shout out to them.

Additionally, bandits now become a latejoinable job role when it's a proper bandit round - if the server rolls for bandits and none are chosen roundstart, players can join later through the latejoin menu. Latejoin slots are capped at what the bandit cap normally is. Bandit migrant waves also still exist and functional normally. 

This is one of my bigger PRs I've done before so there is a possibility of some minor bugs here and there, but I've done my best to test it. Thanks a bunch to Sigma, without them and their support this PR wouldn't be possible. 

### Classes:
_it takes forever to test this and i have no good debug tools so no i didnt take a picture of each class. Sorry!!!_
**Brigand** - axe or club strongman.
image
**Sellsword**- sword mercenary or spear deserter.

**Knave** - sneaky bastard. Comes in no-good thief with lockpicks and poacher with mantraps varieties. 
image
**Hedge Knight** - disgraced noble, fallen from grace. Near identical to the black knight errant. 
image
~~calls it a draw if you cut off all his limbs~~
**Rogue Mage** - those fools at the academy don't know who they're messing with! Like adventurer mage, but more evil.
image
**Sawbones** - Have you considered signing up for the bandit medical plan? No, there's no dental. Stop asking. 
image

### Updated Bandit Camp
image
image

## Why It's Good For The Game

What if our bad guys didn't suck. Better player expression. At the moment bandit 'classes' are chosen by a randomizer - it's hard to roleplay a character when you don't get to select their archetype. This allows antagonist characters to be better fleshed out and have more coherent identities, allowing for a superior roleplay experience. Also will help them not immediately fold over to the first adventurer or guard they encounter. Having them be late joinable also allows people to play dedicated bandit characters, instead of having their normal town role character suddenly be a bandit one day. 

Bandits are historically meant to be a minor antag, but as it says in our code commends a lot of our major antags are disabled as we await custom ones, so bandits are currently fulfilling the role of a major antag with the tools and presence of a minor one. Hopefully these changes allow bandits to be both more meaningful to play as and against.

### CODE REVIEW

**Comment by bearrrrrrrr** [comment]
images


"BANDIT steals the show!"

**Comment by bearrrrrrrr** [comment]
not a fan of it being a job for multiple reasons

1. we lack comprehensive job-ban features, out of the gate
2. this means that it can spawn _**regardless**_ of mode; sometimes it is good for us to put on extended instead of roguemode
3. static hostile spawns do not work. see hearth's vikings on implementation. it's literally the same round every round
4. biggest reason, _**as an admin, frankly, i do not trust our playerbase at present to handle this responsibly.**_ i'd like to give it a while for us to weed out the biggest problem players to avoid a headache.

**Comment by bearrrrrrrr** [comment]
putting this here and in discord; peep code\modules\antagonists\roguetown\villain\vampirelord.dm to see how it does after-spawn class select

File: code/modules/jobs/job_types/roguetown/adventurer/types/antag/brigand.dm [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/datum/advclass/brigand //Strength class, starts with axe or flails and medium armor training
+	name = "Brigand"
+	tutorial = "Cast from society, you use your powerful physical might and endurance to take from those who are weaker from you."
+	allowed_sexes = list(MALE, FEMALE)
+	allowed_races = RACES_ALL_KINDS
+	outfit = /datum/outfit/job/roguetown/bandit/brigand
+	category_tags = list(CTAG_BANDIT)
+
```
Comment (crunchypretzels) [inline code/modules/jobs/job_types/roguetown/adventurer/types/antag/brigand.dm:8]: may i suggest for the melee classes

cmode_music = 'sound/music/combat_bandit.ogg'

File: code/modules/jobs/job_types/roguetown/adventurer/types/antag/knave.dm [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/datum/advclass/knave //sneaky bastards - ranged classes of two flavors archers and rogues
+	name = "Knave"
+	tutorial = "Not all followers of Matthios take by force. Thieves, poachers, and ne'er-do-wells of all forms steal from others from the shadows, long gone before their marks realize their misfortune."
+	allowed_sexes = list(MALE, FEMALE)
+	allowed_races = RACES_ALL_KINDS
+	outfit = /datum/outfit/job/roguetown/bandit/knave
+	category_tags = list(CTAG_BANDIT)
+
```
Comment (crunchypretzels) [inline code/modules/jobs/job_types/roguetown/adventurer/types/antag/knave.dm:8]: and for the ranged...

cmode_music = 'sound/music/combat_bandit2.ogg'

**Review by crunchypretzels — CHANGES_REQUESTED** [review]
hello coder. i need 400 lines of extra code pleab. glory to gitub

File: code/modules/jobs/job_types/roguetown/adventurer/types/antag/brigand.dm (hunk 1 above)
Comment (crunchypretzels) [reply code/modules/jobs/job_types/roguetown/adventurer/types/antag/brigand.dm:8]: while a cool track it doesn't compare to the waltzing melody of Deus Ex - Battery Park (Combat). the building tempo as it grows more frenetic, the marching beat, the ebb and flow of the horns and brass...

**Comment by coffeecoffeecoffee** [comment]
> not a fan of it being a job for multiple reasons
> 
>     1. we lack comprehensive job-ban features, out of the gate
> 
>     2. this means that it can spawn _**regardless**_ of mode; sometimes it is good for us to put on extended instead of roguemode
> 
>     3. static hostile spawns do not work. see hearth's vikings on implementation. it's literally the same round every round
> 
>     4. biggest reason, _**as an admin, frankly, i do not trust our playerbase at present to handle this responsibly.**_ i'd like to give it a while for us to weed out the biggest problem players to avoid a headache. it is very easy to say 'simply ban them', but understand that literally every single ban or note comes with about 30 minutes of review and log-diving and pm'ing folk, potentially far more
> 
> 
> (otherwise, no objections/notes)

1) i'm pretty sure we just need to port it from ratwood; somebody correct me if I'm wrong but they implemented this, and we can see about porting it very easily if so
2) almost every round is basically extended. i, both as a player and admin, am really tired of it. it makes the game lifeless and the struggle feel pointless. we need more antagonists badly.
3) this may be true; i do feel like there's something to the notion that being able to play an antagonist "character" consistently every round all round has turned out unhealthy in the past, and perhaps we could solve this by having this BE a gamemodespawn that opens up a bunch of bandit "slots" for latejoining rather than being consistently constantly available.
4) we have weeded out LOTS of major problem players, particularly bandit ones, if you recall. this is something we should be doing anyways regardless, and the fact is, right now, we are resting on our laurels and getting used to a lifeless, pointless existence languishing in town every round - and cool things will "wait on us to do X" forever. weeding out problem players is not an achievable goal, it is a constant cleaning task.

**Comment by bearrrrrrrr** [comment]
> > not a fan of it being a job for multiple reasons
> > ```
> > 1. we lack comprehensive job-ban features, out of the gate
> > 
> > 2. this means that it can spawn _**regardless**_ of mode; sometimes it is good for us to put on extended instead of roguemode
> > 
> > 3. static hostile spawns do not work. see hearth's vikings on implementation. it's literally the same round every round
> > 
> > 4. biggest reason, _**as an admin, frankly, i do not trust our playerbase at present to handle this responsibly.**_ i'd like to give it a while for us to weed out the biggest problem players to avoid a headache. it is very easy to say 'simply ban them', but understand that literally every single ban or note comes with about 30 minutes of review and log-diving and pm'ing folk, potentially far more
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > (otherwise, no objections/notes)
> 
>     1. i'm pretty sure we just need to port it from ratwood; somebody correct me if I'm wrong but they implemented this, and we can see about porting it very easily if so
> 
>     2. almost every round is basically extended. i, both as a player and admin, am really tired of it. it makes the game lifeless and the struggle feel pointless. we need more antagonists badly.
> 
>     3. this may be true; i do feel like there's something to the notion that being able to play an antagonist "character" consistently every round all round has turned out unhealthy in the past, and perhaps we could solve this by having this BE a gamemodespawn that opens up a bunch of bandit "slots" for latejoining rather than being consistently constantly available.
> 
>     4. we have weeded out LOTS of major problem players, particularly bandit ones, if you recall. this is something we should be doing anyways regardless, and the fact is, right now, we are resting on our laurels and getting used to a lifeless, pointless existence languishing in town every round - and cool things will "wait on us to do X" forever. weeding out problem players is not an achievable goal, it is a constant cleaning task.

'Almost every round is basically extended'? We have so many antags, some days. People aren't literally constantly in combat, sure, but they're not _constantly safe_ and bored. We have complaints that there's **too much going on** some rounds as-is.

Introducing FO13-style outlaws is not the play. Maybe we make bandit migrant waves more common? Or, honestly, WL'ing people for a bandit role would also be fine tbh. I 100% do not think it is a good idea to say 'yes you can drop in as an antag literally whenever you want' to everybody.

**Comment by EvenInDeathIStillServe** [comment]
Definitely possible to make the gamemode open up bandit slots in addition to assigning people as bandits roundstart.

**Comment by AdamElTablawy** [comment]
> > > not a fan of it being a job for multiple reasons
> > > ```
> > > 1. we lack comprehensive job-ban features, out of the gate
> > > 
> > > 2. this means that it can spawn _**regardless**_ of mode; sometimes it is good for us to put on extended instead of roguemode
> > > 
> > > 3. static hostile spawns do not work. see hearth's vikings on implementation. it's literally the same round every round
> > > 
> > > 4. biggest reason, _**as an admin, frankly, i do not trust our playerbase at present to handle this responsibly.**_ i'd like to give it a while for us to weed out the biggest problem players to avoid a headache. it is very easy to say 'simply ban them', but understand that literally every single ban or note comes with about 30 minutes of review and log-diving and pm'ing folk, potentially far more
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > (otherwise, no objections/notes)
> > 
> > 
> > ```
> > 1. i'm pretty sure we just need to port it from ratwood; somebody correct me if I'm wrong but they implemented this, and we can see about porting it very easily if so
> > 
> > 2. almost every round is basically extended. i, both as a player and admin, am really tired of it. it makes the game lifeless and the struggle feel pointless. we need more antagonists badly.
> > 
> > 3. this may be true; i do feel like there's something to the notion that being able to play an antagonist "character" consistently every round all round has turned out unhealthy in the past, and perhaps we could solve this by having this BE a gamemodespawn that opens up a bunch of bandit "slots" for latejoining rather than being consistently constantly available.
> > 
> > 4. we have weeded out LOTS of major problem players, particularly bandit ones, if you recall. this is something we should be doing anyways regardless, and the fact is, right now, we are resting on our laurels and getting used to a lifeless, pointless existence languishing in town every round - and cool things will "wait on us to do X" forever. weeding out problem players is not an achievable goal, it is a constant cleaning task.
> > ```
> 
> 'Almost every round is basically extended'? We have so many antags, some days. People aren't literally constantly in combat, sure. That's not extended. There is so much happening. We have complaints that there's **too much going on** some rounds as-is. It may be worth looking at how antags spawn, but, that being said-
> 
> Introducing FO13-style outlaws is not the play. Maybe we make bandit migrant waves more common? Or, honestly, WL'ing people for a bandit role would also be fine tbh. I 100% do not think it is a good idea to say 'yes you can drop in as an antag literally whenever you want' to everybody.

Difference of opinion, honestly. Most of my friends complain there's not really a lot going on. You'll typically get one antag per round, or two, and they get dumpstered fast. A lot of rounds really aren't that chaotic. (If I was being uncharitable, i'd say that a lot of the people saying there's 'too much' come from Hearthstone or Nova and are accustomed to literally nothing ever.)

All the same, there's no harm in testmerging this for a week or so and seeing how it goes.

**Comment by chemistrymain3** [comment]
> 'Almost every round is basically extended'? We have so many antags, some days. People aren't literally constantly in combat, sure. That's not extended. There is so much happening. We have complaints that there's too much going on some rounds as-is. It may be worth looking at how antags spawn, but, that being said-

Difference of opinion, honestly. Most of my friends complain there's not really a lot going on. You'll typically get one antag per round, or two, and they get dumpstered fast. A lot of rounds really aren't that chaotic. (If I was being uncharitable, i'd say that a lot of the people saying there's 'too much' come from Hearthstone or Nova and are accustomed to literally nothing ever.)

All the same, there's no harm in testmerging this for a week or so and seeing how it goes.

**Comment by chemistrymain3** [comment]
> > Difference of opinion, honestly. Most of my friends complain there's not really a lot going on. You'll typically get one antag per round, or two, and they get dumpstered fast. A lot of rounds really aren't that chaotic. (If I was being uncharitable, i'd say that a lot of the people saying there's 'too much' come from Hearthstone or Nova and are accustomed to literally nothing ever.)
> > All the same, there's no harm in testmerging this for a week or so and seeing how it goes.
> 
> I mean, this is also a case of 'nobody rolls antags' some rounds, tbf. Yesterday and the day before I did a fun little experiment where I announced really loudly that antags were on and to pretty please enable them in people's prefs.
> 
> We got zero antag rollers.
> 
> This rots.
> 
> I really like the idea of a latejoin being opened up when bandits roll, though. That's dope :3

A big reason nobody rolls antags is that rolling antag means that you are one guy against the town's armada in full plate.

Being able to join as a bandit with a couple other bandits reliably would help resolve this.

**Comment by bearrrrrrrr** [comment]
just to confirm, as with the PR desc, this is a job that opens up after bandits roll for the round? if so that's awesome, great work :3

**Pull request merged**